core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -199,21 +198,29 @@ private void initializeCatalogTables() throws InterruptedException, SQLException
         });
   }
 
-  private void updateCatalogTables(Connection connection) throws SQLException {
-    DatabaseMetaData dbMeta = connection.getMetaData();
-    ResultSet typeColumn =
-        dbMeta.getColumns(null, null, JdbcUtil.CATALOG_TABLE_VIEW_NAME, JdbcUtil.RECORD_TYPE);
-    if (typeColumn.next()) {
-      LOG.debug("{} already supports views", JdbcUtil.CATALOG_TABLE_VIEW_NAME);
-    } else {
-      if (PropertyUtil.propertyAsBoolean(
-          catalogProperties, JdbcUtil.ADD_VIEW_SUPPORT_PROPERTY, false)) {
-        connection.prepareStatement(JdbcUtil.UPDATE_CATALOG_SQL).execute();
-      } else {
-        LOG.warn(VIEW_WARNING_LOG_MESSAGE);
-        schemaVersion = JdbcUtil.SchemaVersion.V0;
-      }
-    }
+  private void checkAndMaybeUpdateSchema() throws SQLException, InterruptedException {
```

Comment:
Nit: It feels a bit weird to have "maybe" in a method name like this, could we call this `updateSchemaIfRequired`

Reply:
OK, no problem. I don't mind to change the name especially for a `private` method 😄

Reply:
+1 on `updateSchemaIfRequired`